### ISSUE TITLE
rpc error: code = Unavailable desc = transport is closing ------> with route53 resource

### ISSUE THREAD

**Issue by ram-starlordtech** [terraform-providers/terraform-provider-aws#9910]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* aws_XXXXX

### Terraform Configuration Files

```hcl
resource "aws_alb_listener" "sites-startlordtech-listener-HTTPS" {

  load_balancer_arn = "${aws_lb.ALB.arn}"
  port              = "443"
  protocol          = "HTTPS"
  ssl_policy = "ELBSecurityPolicy-2016-08"
  certificate_arn = "${var.sslcertificate-arn-starlordtech}"

  default_action {
    type = "fixed-response"

    fixed_response {
      content_type = "text/plain"
      message_body = "Fixed response content"
      status_code  = "503"
    }
  }
}
//data "aws_instances" "test_instances" {
//}
//
//
resource "aws_alb_target_group_attachment" "starlordtech-target-group-instancestest" {
   //count            = 3
   target_group_arn = "${aws_alb_target_group.app.arn}"
   target_id = "${aws_instance.Staging_instance.id}"
    port = 80

}


resource "aws_route53_zone" "primary_Staging" {
  name = "starlordtech.Staging.internal"
}


resource "aws_route53_record" "staging_TF_record1" {
  zone_id = "${aws_route53_zone.primary_Staging.zone_id}"
  name    = "staging3."
  type    = "A"
  ttl     = "300"
  records = ["${aws_instance.Staging_instance.public_ip}"]
}


resource "aws_route53_record" "staging_TF_record2" {
  zone_id = "${aws_route53_zone.primary_Staging.zone_id}"
  name    = "stagingSPOT."
  type    = "A"
  ttl     = "300"
  records = ["${aws_spot_instance_request.cheap_worker.public_ip}"]
}```

### Debug Output


<script src="https://gist.github.com/ram-starlordtech/b730ffb8ffb0dda69c82aadaa7483001.js"></script>


### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behavior

Created resources

### Actual Behavior

<!--- TERRAFORM CRASH --->

### Steps to Reproduce

<!---I cannot send all config files, Every time I try to create fresh all resources. Fresh. its crashing. Main thing or my script create all new and fresh resources not updating.  It started after I added route 53 resources. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

**Comment by bflad** [comment]
Hi @ram-starlordtech 👋 Thank you for reporting this and sorry you ran into trouble here.

Given the panic output (provided below), this crash appears to be occurring the Terraform Provider SDK that currently lives upstream in the Terraform codebase. The only change that would potentially need to happen in this case would be updating our dependency if a fix was merged upstream. To follow further updates upstream, a relevant issue appears to be: https://github.com/hashicorp/terraform/issues/22218

I believe we have seen similar panics that have been fixed though, so you may want to see if upgrading to the latest Terraform (0.12.7) and Terraform AWS Provider (2.25.0) also resolves this, as both places may have addressed this already.

```
2019-08-29T15:20:00.741+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: panic: reflect: call of reflect.Value.Type on zero Value
2019-08-29T15:20:00.741+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: 
2019-08-29T15:20:00.741+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: goroutine 762 [running]:
2019-08-29T15:20:00.741+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: reflect.Value.Type(0x0, 0x0, 0x0, 0x38, 0x200)
2019-08-29T15:20:00.741+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: 	/opt/goenv/versions/1.12.6/src/reflect/value.go:1813 +0x169
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.validatePrimitive(0xc0005beb40, 0xc0008e0160, 0x9, 0x0, 0x0, 0xc0005de300, 0xc0006c3230, 0x1, 0x492c720, 0xc000ac6420, ...)
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:1694 +0xc8
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.validateType(0xc0005beb40, 0xc0008e0160, 0x9, 0x0, 0x0, 0xc0005de300, 0xc0006c3230, 0x49a24a0, 0x49a24a0, 0x100f701, ...)
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:1779 +0x56f
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.validateList(0xc0005beb40, 0x579f000, 0x7, 0x492c720, 0xc000ac6420, 0xc0005df100, 0xc0006c3230, 0xc0006c3260, 0x0, 0x0, ...)
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:1495 +0x86c
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.validateType(0xc0005beb40, 0x579f000, 0x7, 0x492c720, 0xc000ac6420, 0xc0005df100, 0xc0006c3230, 0x0, 0x0, 0x0, ...)
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:1775 +0x9e
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.validate(0xc0005beb40, 0x579f000, 0x7, 0xc0005df100, 0xc0006c3230, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:1382 +0x213
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.validateObject(0xc0005beb40, 0x0, 0x0, 0xc0005beb40, 0xc0006c3230, 0x57c2668, 0xc0008060c0, 0x12, 0xc000408401, 0xc000000000000000, ...)
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:1661 +0x1c6
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.Validate(...)
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:677
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Validate(0xc0005c3380, 0xc0006c3230, 0xc0008060c0, 0x12, 0xc000664798, 0x1, 0x40, 0x0)
2019-08-29T15:20:00.742+0800 [DEBUG] plugin.terraform-provider-aws_v2.22.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:357 +0x5e
```

**Comment by bflad** [comment]
Upon further investigation, this should be resolved by https://github.com/hashicorp/terraform/pull/22236, which was merged into the Terraform AWS Provider via https://github.com/terraform-providers/terraform-provider-aws/pull/9577 and released in version 2.23.0. Upgrading the Terraform AWS Provider version 2.23.0 or later should resolve the issue. 👍 

If you continue to have trouble, please feel free to open a new [Bug Report](https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/ISSUE_TEMPLATE/Bug_Report.md) with any updated logs and we'll investigate further. Thanks!
